crypto/elliptic-curve.ts

```diff
@@ -703,3 +716,369 @@ function createCurveAffine({
     },
   };
 }
+
+/**
+ * Parameters defining an elliptic curve in twisted Edwards form
+ * ax^2 + y^2 = 1 + dx^2y^2
+ */
+type TwistedCurveParams = {
+  /**
+   * Human-friendly name for the curve
+   */
+  name: string;
+  /**
+   * Base field modulus
+   */
+  modulus: bigint;
+  /**
+   * Scalar field modulus = group order
+   */
+  order: bigint;
+  /**
+   * Cofactor = size of EC / order
+   *
+   * This can be left undefined if the cofactor is 1.
+   */
+  cofactor?: bigint;
+  /**
+   * Generator point
+   */
+  generator: { x: bigint; y: bigint };
+  /**
+   * The `a` parameter in the curve equation ax^2 + y^2 = 1 + dx^2y^2
+   */
+  a: bigint;
+  /**
+   * The `d` parameter in the curve equation ax^2 + y^2 = 1 + dx^2y^2
+   */
+  d: bigint;
+  endoBase?: bigint;
+  endoScalar?: bigint;
+};
+
+function twistedOnCurve(
+  { x, y, infinity }: GroupTwisted,
+  p: bigint,
+  a: bigint,
+  d: bigint
+) {
+  if (infinity) return true;
+  // a * x^2 + y^2 = 1 + d * x^2 * y^2
```

Comment:
I guess we want to check if it is in the prime subgroup too? If yes, I would create a second method instead of adding in `isOnCurve`.